### PR TITLE
Feature/zip api

### PR DESCRIPTION
**Pull request by disconsented**

### CODE REVIEW

**Comment by disconsented** [comment]
I'll just write this here to keep track of it:

- [ ] Filter for file extensions on the API endpoint to simplify logic for the front end
- [ ] Add support to use archive/extract as an operation (used in templates)

I am pretty open to most changes at this point, anything else you'd like to add?

**Comment by disconsented** [comment]
Ends up being that the library already filters extensions for us

https://github.com/mholt/archiver/blob/074da64920d3c1188f608a92eceb61e83b31317b/archiver.go#L438

**Comment by disconsented** [comment]
Not totally sure how the frontend is made aware of operations but other than that this is good to go

File: web/daemon/server.go [hunk 1]
```diff
@@ -668,6 +672,60 @@ func GetStatus(c *gin.Context) {
...
+// @Param id path string true "Server Identifier"
+// @Param filename path string true "File name"
+// @Param destination path string true "Destination directory (URI Parameter)"
+// @Router /daemon/server/{id}/extract/{filename} [get]
+func Extract(c *gin.Context) {
+	item, _ := c.Get("server")
+	server := item.(*programs.Program)
+
+	targetPath := c.Param("filename")
+	destination := c.Query("destination")
```
Comment (LordRalex) [inline web/daemon/server.go:720]: No checks are done to ensure destination isn't empty, that is a client error and should be reported as such

File: web/daemon/server.go [hunk 2]
```diff
@@ -668,6 +672,60 @@ func GetStatus(c *gin.Context) {
...
+// @Param filename path string true "Destination"
+// @Router /daemon/server/{id}/archive/{filename} [post]
+func Archive(c *gin.Context) {
+	item, _ := c.Get("server")
+	server := item.(*programs.Program)
+	var files []string
+	if err := c.BindJSON(&files); response.HandleError(c, err, http.StatusBadRequest) {
+		return
+	}
+	destination := c.Param("filename")
```
Comment (LordRalex) [inline web/daemon/server.go:693]: If we have no files to zip up, should be a 400 error

**Review by LordRalex — APPROVED** [review]

**Comment by LordRalex** [comment]
Needs to be rebased, since the go.mod changed in master with the other PR

**Comment by disconsented** [comment]
@LordRalex should be safe to merge now

**Pull request merged**